tests/baselines/reference/indexedAccessRelation.errors.txt

```diff
@@ -22,13 +24,14 @@ tests/cases/compiler/indexedAccessRelation.ts(16,25): error TS2322: Type 'T' is
     {
         foo(a: T) {
             this.setState({ a: a });
-                            ~
-!!! error TS2322: Type 'T' is not assignable to type 'S["a"] & T'.
-!!! error TS2322:   Type 'Foo' is not assignable to type 'S["a"] & T'.
-!!! error TS2322:     Type 'Foo' is not assignable to type 'S["a"]'.
-!!! error TS2322:       Type 'T' is not assignable to type 'S["a"]'.
-!!! error TS2322:         Type 'Foo' is not assignable to type 'S["a"]'.
-!!! related TS6500 tests/cases/compiler/indexedAccessRelation.ts:8:5: The expected type comes from property 'a' which is declared here on type 'Pick<S & State<T>, "a">'
+                          ~~~~~~~~
+!!! error TS2345: Argument of type '{ a: T; }' is not assignable to parameter of type 'Pick<S & State<T>, "a">'.
+!!! error TS2345:   Types of property 'a' are incompatible.
+!!! error TS2345:     Type 'T' is not assignable to type 'S["a"] & T'.
+!!! error TS2345:       Type 'Foo' is not assignable to type 'S["a"] & T'.
+!!! error TS2345:         Type 'Foo' is not assignable to type 'S["a"]'.
+!!! error TS2345:           Type 'T' is not assignable to type 'S["a"]'.

```

Comment:
Seems a little weird to have this extra line, and it's supposedly because intersections always have an extra `this`-type? Seems like a leak of impl. details

Reply:
Should open another issue for this; this was the prior message.